### PR TITLE
yanli dong

### PR DESCRIPTION
**Pull request by dongyanlihello**

### CODE REVIEW

**Comment by dongyanlihello** [comment]
I have submitted my code for the challenge. Thank you.

**Comment by CharlesAHunt** [comment]
Thank you for submitting your solution @dongyanlihello  . We will be adding comments to your PR shortly. Keep a watch on this repository for the next couple days so you can respond to your reviewer's comments. 👍

**Comment by dongyanlihello** [comment]
Thank you.

Charles Hunt <notifications@github.com> 于2019年11月10日周日 下午11:33写道：

File: Back End Challenge/pom.xml [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    </dependency>
+    <dependency>
+      <groupId>javax.servlet</groupId>
+      <artifactId>servlet-api</artifactId>
+      <version>2.5</version>
+    </dependency>
+
+    <!-- framework -->
+    <dependency>
+      <groupId>org.springframework</groupId>
```
Comment (CharlesAHunt) [inline Back End Challenge/pom.xml]: How did you decide on using the Spring Framework to solve this challenge? 
What are the advantages, and very importantly, what are the disadvantages to using the Spring Framework in relation to other choices?

File: Back End Challenge/src/main/java/com/test/headstorm/challenge/controller/RandomNumbersController.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import javax.annotation.Resource;
+import java.util.List;
+
+/**
+ * <p>  </p>
+ *
+ * @author
+ * @version $Id: RandomNumbersController.java, v 0.1 2019/8/15  Exp $
+ */
+@RestController
```
Comment (CharlesAHunt) [inline Back End Challenge/src/main/java/com/test/headstorm/challenge/controller/RandomNumbersController.java]: Are these annotations checked and used at compile-time or run-time?

File: Back End Challenge/src/main/java/com/test/headstorm/challenge/controller/RandomNumbersController.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    RandomNumbersService randomNumbersService;
+
+    @PostMapping(value="/numbers", consumes = "application/json")
+    public String numbersCheck(@RequestBody List<Integer> reqList) {
+        return randomNumbersService.numbersCheck(reqList);
+    }
+
+    @GetMapping(value="/numbers", consumes = "application/json")
+    public List<Integer> numbersSort(@RequestBody List<Integer> reqList) {
+        return randomNumbersService.numbersSort(reqList);
```
Comment (CharlesAHunt) [inline Back End Challenge/src/main/java/com/test/headstorm/challenge/controller/RandomNumbersController.java]: If we discover that this data service is very "read-heavy", or in other words, that our users POST new lists **very** rarely, but GET the number list **very** frequently, how would you modify or optimize this solution to perform less work and return faster responses to your users in this case?

File: Back End Challenge/pom.xml [hunk 4]
```diff
@@ -0,0 +1,205 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (CharlesAHunt) [inline Back End Challenge/pom.xml]: If you could choose any other build tool besides Maven for Java, what would you choose and why?

**Comment by CharlesAHunt** [comment]
Thanks for your responses @{user} ! This wraps up the Github challenge portion. Our recruiting team will be in contact with next steps as soon as possible. Thanks! 👍

**Comment by CharlesAHunt** [comment]
@dongyanlihello We have added our questions to your PR. As soon as you complete the answers, we can wrap up this portion of the interview process and move forward. :-)

**Comment by CharlesAHunt** [comment]
Hey @dongyanlihello. Just checking in to see if you we're still interested in completing the PR questions. We'd love to hear back from you! :-)

**Pull request closed without merging**